### PR TITLE
chore(docs): Documenting configuration apis

### PR DESCRIPTION
**Pull request by lukeblevins**
GitHub Issue (If applicable): #1531 

Closing #1531 

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

This PR contains a breaking change (ed6f27eb902667b9945c07708be6aaa0182674f0) which I'm seeking approval for.

## Other information

Internal Issue (If applicable): N/A

### CODE REVIEW

File: testing/TestHarness/TestHarness.Shared/BaseHostInitialization.cs [hunk 1]
```diff
@@ -67,11 +67,11 @@ protected virtual IHostBuilder Configuration(IHostBuilder builder)
 			// Only use this syntax for UI tests - use UseConfiguration in apps
 			builder = builder.ConfigureAppConfiguration((ctx, b) =>
 			{
-				b.AddEmbeddedConfigurationFile<App>(file);
+				b.AddEmbeddedConfiguration<App>(ctx, file);
```
Comment (nickrandolph) [inline testing/TestHarness/TestHarness.Shared/BaseHostInitialization.cs:70]: Why was this changed?

Comment (lukeblevins) [reply testing/TestHarness/TestHarness.Shared/BaseHostInitialization.cs:70]: I did that to reflect commit 0e8240c71f922ea78678261f61d4ac9b2d8eae99 which marked `AddEmbeddedConfigurationFile<T>` `private`. That method appeared to have been exposed by mistake, but we did invoke it from test harness. The change you noticed above updated the test to reference only a `public` method.

File: src/Uno.Extensions.Configuration/ConfigBuilderExtensions.cs [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
 	public const string ConfigurationFolderName = "config";
+
+	/// <summary>
+	/// Sets up the host builder to register content files by name as a configuration source
+	/// </summary>
+	/// <param name="hostBuilder">
+	/// The <see cref="IConfigBuilder"/> to configure.
+	/// </param>
+	/// <param name="config">
+	/// A name to identify the added configuration source. Optional
```
[the remark this answers is not in the document]
Comment (lukeblevins) [reply src/Uno.Extensions.Configuration/ConfigBuilderExtensions.cs]: @nickrandolph Yep, the string provided `foo` should actually be a _name for a file_. Though, I wouldn't call it a _file name_ given that this value ends up being formatted to fit the template 😆
```csharp
namespace Uno.Extensions.Configuration;

internal static class AppConfiguration
{
        ...
	public const string FileNameTemplate = $"{Prefix}.{{0}}.json";
}
```

On a technicality, the formatted string becomes a _file name_
`appsettings.foo.json`

Is there a more specific term than _file name_ we can use here? When I hear _file name_ I assume it should contain an extension, when we instead are asking only for a suffix or identifier to fit our template.

File: src/Uno.Extensions.Configuration/ConfigBuilderExtensions.cs [hunk 3]
```diff
@@ -52,6 +94,21 @@ public static IConfigBuilder EmbeddedSource<TApplicationRoot>(this IConfigBuilde
...
+	/// The type that the configuration section will be deserialized to.
+	/// </typeparam>
+	/// <param name="hostBuilder">
+	/// The <see cref="IConfigBuilder"/> to configure.
+	/// </param>
+	/// <param name="configurationSection">
+	/// The configuration section to retrieve.
+	/// </param>
+	/// <returns>
+	/// The same instance of the <see cref="IConfigBuilder"/> for chaining.
```
[the remark this answers is not in the document]
Comment (lukeblevins) [reply src/Uno.Extensions.Configuration/ConfigBuilderExtensions.cs]: Fixed!

File: src/Uno.Extensions.Configuration/ConfigurationBuilderExtensions.cs [hunk 4]
```diff
@@ -1,43 +1,73 @@
...
+	/// <param name="configurationBuilder">The builder instance used to create an <see cref="IConfiguration"/> with keys and values from a set of sources</param>
+	/// <param name="hostingContext">The <see cref="HostBuilderContext"/> which provides information specific to the hosting environment</param>
+	/// <returns>The <see cref="IConfigurationBuilder"/>.</returns>
 	public static IConfigurationBuilder AddEnvironmentAppConfiguration(this IConfigurationBuilder configurationBuilder, HostBuilderContext hostingContext)
 	{
 		return configurationBuilder.AddEnvironmentConfiguration(hostingContext, default);
 	}
 
-	public static IConfigurationBuilder AddEmbeddedConfigurationFile<TApplicationRoot>(this IConfigurationBuilder configurationBuilder, string configurationFileName)
+	private static IConfigurationBuilder AddEmbeddedConfigurationFile<TApplicationRoot>(this IConfigurationBuilder configurationBuilder, string configurationFileName)
```
Comment (nickrandolph) [inline src/Uno.Extensions.Configuration/ConfigurationBuilderExtensions.cs:70]: This needs to be public as there needs to be a way that you can specify the exact configuration file you want to load from.
AddEmbeddedConfiguration 
- if configurationName is null, will use appsettings.json instead
- if configurationName is not null, will use appsettings.configurationName.json

AddEmbeddedConfigurationFile will use the file called configurationFileName (exactly as it's provided)

Comment (lukeblevins) [reply src/Uno.Extensions.Configuration/ConfigurationBuilderExtensions.cs]: I've addressed this with the latest commits and annotated this method with the behavior described above

File: testing/TestHarness/TestHarness.Shared/BaseHostInitialization.cs (hunk 1 above)
Comment (lukeblevins) [reply testing/TestHarness/TestHarness.Shared/BaseHostInitialization.cs]: Done!

**Review by nickrandolph — APPROVED** [review]
